File: python-pr/52/python/code-quality-performance/string_matcher_08.py
# {fact rule=code-quality-performance@v1.0 defects=1}
def contains_substring(text, pattern):
    for i in range(len(text) - len(pattern) + 1):
        match = True
        for j in range(len(pattern)):
            if text[i + j] != pattern[j]:
                match = False
                break
        if match:
            return True
    return False
# {/fact}

File: js-pr/56/javascript/code-quality-performance/string_matcher_08.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function containsSubstring(text, pattern) {
    for (let i = 0; i <= text.length - pattern.length; i++) {
        let match = true;
        for (let j = 0; j < pattern.length; j++) {
            if (text[i + j] !== pattern[j]) {
                match = false;
                break;
            }
        }
        if (match) return true;
    }
    return false;
}
// {/fact}